src/Interpreters/InterpreterCreateQuery.cpp

```diff
@@ -910,66 +911,13 @@ void InterpreterCreateQuery::validateTableStructure(const ASTCreateQuery & creat
 
     const auto & settings = getContext()->getSettingsRef();
 
-    /// Check low cardinality types in creating table if it was not allowed in setting
-    if (!create.attach && !settings.allow_suspicious_low_cardinality_types && !create.is_materialized_view)
+    /// If it's not attach and not materialized view to existing table,
+    /// we need to validate data types (check for experimental or suspicious types).
+    if (!create.attach && !create.is_materialized_view)
```

Comment:
In the old version `!create.is_materialized_view` was only used in the first condition. I think the new version makes more sense, but I am not sure if you intended like this or just didn't noticed.

Reply:
It was intended, we should have this condition for all checks